lib/improver/ensemble_copula_coupling/ensemble_copula_coupling.py

```diff
@@ -622,22 +622,100 @@ def process(self, forecast_probabilities, no_of_percentiles=None,
         return forecast_at_percentiles
 
 
-class GeneratePercentilesFromMeanAndVariance(BasePlugin):
+class FromMeanAndVariance():
+    """
+    Base Class to support the plugins that compute percentiles and
+    probabilities from the mean and variance.
+    """
+
+    def __init__(self, distribution="norm", shape_parameters=None):
+        """
+        Initialise the class.
+
+        Args:
+            distribution (str):
+                Name of a distribution supported by scipy.stats.
+            shape_parameters (list or None):
+                For use with distributions in scipy.stats (e.g. truncnorm) that
+                require the specification of shape parameters to be able to
+                define the shape of the distribution.
+                Please note that for use with
+                :meth:`~improver.ensemble_calibration.ensemble_calibration.\
+ContinuousRankedProbabilityScoreMinimisers.calculate_truncated_normal_crps`,
+                the shape parameters for a truncated normal distribution with
+                a lower bound of zero should be [0, np.inf].
+
+        """
+        try:
+            self.distribution = getattr(stats, distribution)
+        except AttributeError as err:
+            msg = ("The distribution requested {} is not a valid distribution "
+                   "in scipy.stats. {}".format(distribution, err))
+            raise AttributeError(msg)
+
+        if shape_parameters is None:
+            shape_parameters = []
+        self.shape_parameters = shape_parameters
+
+    def __repr__(self):
+        """Represent the configured plugin instance as a string."""
+        result = ('<FromMeanAndVariance: distribution: {}; '
+                  'shape_parameters: {}>')
+        return result.format(
+            self.distribution.name, self.shape_parameters)
+
+    def _rescale_shape_parameters(self, mean, std):
+        """
+        Rescale the shape parameters for the desired mean and standard
```

Comment:
As discussed, it is not immediately obvious how the shape parameters work for the scipy.stats.truncnorm function. We've had a play around and now understand how it works. I'm not sure how to make it clear in this docstring, but feel free to have a go.

Reply:
We probably need to make it clear in the top level docstring that the scaling is applied to the shape parameters for truncated normal, so the shape parameters given should be where they want the shifted distribution to be truncated at.

Reply:
I've tried to add a little bit detail to this docstring. As we discussed earlier, the reason for the rescaling is because otherwise the shape parameters are assumed to be for a standard normal distribution, and as we define a distribution using specific values for the mean and standard deviation, the assumption of a standard normal distribution is not appropriate.